includes/modules/pages/download/header_php.php

```diff
@@ -28,245 +29,183 @@
+}
+
+if ($browser_headers_override != '') {
+  /**
+   * Now send the file with header() magic
+   * The "must-revalidate" and expiry times are used to prevent caching and fraudulent re-acquiring of files w/o redownloading.
+   * Certain browsers require certain header combinations, especially when related to SSL mode and caching
+   */
+
+  /**
+   * Set mime type
+   * These Content-Type headers should cause the downloaded file to trigger a "save as" dialog, instead of opening directly
+   * However, some browsers and some software already installed on the PC may cause this to be overridden simply based on the filename. In these cases, the user will have to manually choose "Save As" themselves.
+   * Alternatively, simply set the right handler in .htaccess
+   */
 //    header("Content-Type: application/x-octet-stream");
 //    header("Content-Type: application/octet-stream");
 //    header("Content-Type: application/download");
-    header("Content-Type: application/force-download");
-
-    header('Content-Disposition: attachment; filename="' . urlencode($browser_filename) . '"');
-
-//     relocated below
-//     if ((int)$downloadFilesize > 0) header("Content-Length: " . (string) $downloadFilesize);
-
-    header("Expires: Mon, 22 Jan 2002 00:00:00 GMT");
-    header("Last-Modified: " . gmdate("D,d M Y H:i:s") . " GMT");
-    switch($detectedBrowser)
-    {
-      case 'IE5':
-      case 'IE6':
-        header("Pragma: public");
-        header("Cache-Control: must-revalidate, post-check=0, pre-check=0");
-        header("Cache-Control: private", FALSE);
-        header("Cache-Control: max-age=1");  // stores for only 1 second, which helps allow SSL downloads to work more reliably in IE
-      break;
-      case 'IE7':
-      case 'IE8':
-      case 'IE9':
-        header("Pragma: public");
-        header("Cache-Control: must-revalidate, post-check=0, pre-check=0");
-        header("Cache-Control: private", FALSE);
-        header("Cache-Control: max-age=1");  // stores for only 1 second, which helps allow SSL downloads to work more reliably in IE
-        break;
-      default:
-        header("Cache-Control: no-cache, must-revalidate");
-        header("Pragma: no-cache");
+//    header("Content-Type: application/force-download");
+  header("Content-Type: " . $mime_type);
+
+  header('Content-Disposition: attachment; filename="' . urlencode($browser_filename) . '"');
+
+  header("Expires: Mon, 22 Jan 2002 00:00:00 GMT");
+  header("Last-Modified: " . gmdate("D,d M Y H:i:s") . " GMT");
+  switch($detectedBrowser)
+  {
+    case 'IE5':
+    case 'IE6':
+      header("Pragma: public");
+      header("Cache-Control: must-revalidate, post-check=0, pre-check=0");
+      header("Cache-Control: private", FALSE);
+      header("Cache-Control: max-age=1");  // stores for only 1 second, which helps allow SSL downloads to work more reliably in IE
+    break;

```

Comment:
As lines 175 - 178 are the same as 183-186, would suggest further refactoring to remove the duplicate code.